src/urllib3/filepost.py

```diff
@@ -77,6 +77,8 @@ def encode_multipart_formdata(
 
         if isinstance(data, str):
             writer(body).write(data)
+        elif isinstance(data, typing.BinaryIO):
+            body.write(data.read())
```

Comment:
This line isn't currently tested, and I need to think more about what the intention is here, because I haven't found a way to trigger this branch.

In particular neither an open file nor an `io.BytesIO` return True here, so I wonder if a different check is what's meant (if this branch is even needed at all?):

```python
import io
import typing

with open(__file__, 'rb') as f:
    print(isinstance(f, typing.BinaryIO))  # False

bytes_io = io.BytesIO(b'Hello, World!')
print(isinstance(bytes_io, typing.BinaryIO))  # False
```

Reply:
Maybe bytearray? I can't remember honestly